src/libstd/os/linux/syscall/x86.rs

```diff
@@ -0,0 +1,115 @@
+#[inline(always)]
+pub unsafe fn syscall0(n: usize) -> usize {
+    let ret : usize;
+    asm!("int $$0x80" : "={eax}"(ret)
+                      : "{eax}"(n)
+                      : "memory" "cc"
+                      : "volatile");
+    ret
+}
+
+#[inline(always)]
+pub unsafe fn syscall1(n: usize, a1: usize) -> usize {
+    let ret : usize;
+    asm!("int $$0x80" : "={eax}"(ret)
+                      : "{eax}"(n), "{ebx}"(a1)
+                      : "memory" "cc"
+                      : "volatile");
+    ret
+}
+
+#[inline(always)]
+pub unsafe fn syscall2(n: usize, a1: usize, a2: usize) -> usize {
+    let ret : usize;
+    asm!("int $$0x80" : "={eax}"(ret)
+                      : "{eax}"(n), "{ebx}"(a1), "{ecx}"(a2)
+                      : "memory" "cc"
+                      : "volatile");
+    ret
+}
+
+#[inline(always)]
+pub unsafe fn syscall3(n: usize, a1: usize, a2: usize, a3: usize) -> usize {
+    let ret : usize;
+    asm!("int $$0x80" : "={eax}"(ret)
+                      : "{eax}"(n), "{ebx}"(a1), "{ecx}"(a2), "{edx}"(a3)
+                      : "memory" "cc"
+                      : "volatile");
+    ret
+}
+
+#[inline(always)]
+pub unsafe fn syscall4(n: usize, a1: usize, a2: usize, a3: usize,
+                                a4: usize) -> usize {
+    let ret : usize;
+    asm!("int $$0x80" : "={eax}"(ret)
+                      : "{eax}"(n), "{ebx}"(a1), "{ecx}"(a2), "{edx}"(a3),
+                        "{esi}"(a4)
+                      : "memory" "cc"
+                      : "volatile");
+    ret
+}
+
+#[inline(always)]
+pub unsafe fn syscall5(n: usize, a1: usize, a2: usize, a3: usize,
+                                a4: usize, a5: usize) -> usize {
+    let ret : usize;
+    asm!("int $$0x80" : "={eax}"(ret)
+                      : "{eax}"(n), "{ebx}"(a1), "{ecx}"(a2), "{edx}"(a3),
+                        "{esi}"(a4), "{edi}"(a5)
+                      : "memory" "cc"
+                      : "volatile");
+    ret
+}
+
+#[inline(always)]
+pub unsafe fn syscall6(n: usize, a1: usize, a2: usize, a3: usize,
+                                a4: usize, a5: usize, a6: usize) -> usize {
+    let ret : usize;
+
+    //
+    // this fails when building without optimizations:
```

Comment:
Would something like `optimize(always)` work here?

Reply:
It could, but since `optimize` attribute is ignored in the presence of `-C opt-level=n`, I am not sure if we should use it. Although it's worth to check if this bug is still present.

Reply:
@newpavlov I'm really surprised that `optimize` is ignored based on command-line options; this is a good use case for it *always* working.

I'd love to avoid this extra overhead if possible, but this doesn't need to be a blocker.

Reply:
I would be concerned if `optimize(...)` would impact whether the build is successful / not. `optimize(...)` is a hint and has zero guarantees.